vscode-trace-extension/src/trace-explorer/opened-traces/trace-explorer-opened-traces-webview-provider.ts

```diff
@@ -16,15 +18,20 @@ const JSONBig = JSONBigConfig({
 export class TraceExplorerOpenedTracesViewProvider implements vscode.WebviewViewProvider {
 
 	public static readonly viewType = 'traceExplorer.openedTracesView';
+	public static readonly eventEmitter_id_callback = this.viewType + '.register_callback';
 
 	private _view?: vscode.WebviewView;
 	private _disposables: vscode.Disposable[] = [];
 	private _selectedExperiment: Experiment | undefined;
 
-   	constructor(
+	constructor(
 		private readonly _extensionUri: vscode.Uri,
 		private readonly _statusService: TraceServerConnectionStatusService,
-	) {}
+	) {
+	    traceExtensionEventManager.registerEventEmitter(TraceExplorerOpenedTracesViewProvider.eventEmitter_id_callback);
+	    traceExtensionEventManager.onEvent(TraceExplorerOpenedTracesViewProvider.eventEmitter_id_callback,
+	        (data: unknown) => {this.registerCallbackForSignal(data as SignalCallbackInterface);});
```

Comment:
These registrations have a somewhat odd lifecycle: they live as long as the `TraceExplorerOpenedTracesViewProvider` - the life of the plugin - but the registrations in `registerCallbackForSignal` apply only to a given webview. Perhaps instead, the API should expose an event that is fired when a new webview is created, and subscribers can register whatever signal handlers they wish directly on the webview, rather than going through this indirect event -> event system?